lib/_tls_common.js

```diff
@@ -137,20 +137,33 @@ exports.createSecureContext = function createSecureContext(options, context) {
   }
 
   if (options.pfx) {
-    var pfx = options.pfx;
-    var passphrase = options.passphrase;
-
     if (!crypto)
       crypto = require('crypto');
 
-    pfx = crypto._toBuf(pfx);
-    if (passphrase)
-      passphrase = crypto._toBuf(passphrase);
-
-    if (passphrase) {
-      c.context.loadPKCS12(pfx, passphrase);
+    if (Array.isArray(options.pfx)) {
+      for (i = 0; i < options.pfx.length; i++) {
+        const pfx = options.pfx[i];
+        let buf;
+        if (ArrayBuffer.isView(pfx.buffer)) {
+          buf = crypto._toBuf(pfx.buffer);
+        } else {
+          buf = crypto._toBuf(pfx);
+        }
```

Comment:
I am not completely happy with the buffer check here.

I think it would be good to verify that the input is indeed a buffer and that does not happen here.

If the object attribute name would not be `buffer` it would be easy to distinguish the object from the buffer and you could write something like:

```js
const raw = pfx.buf ? pfx.buf : pfx;
if (!ArrayBuffer.isView(raw))
  throw new Error("foobar");
const buf = crypto._toBuf(raw);
```

Reply:
Looks as a good proposal. ACK.